### DISCUSSION TITLE
AERIoe plots with ceilometer could base height.

### DISCUSSION THREAD

**Discussion by dennyh-ssec** [ARM-DOE/ACT#612] [Show and tell]
![sgpaerioe1turnC1 c1 20220211](https://user-images.githubusercontent.com/110182485/215130104-6f67cc2b-af71-4f6b-b3da-cb01d5c50739.png)

`import act`
`import matplotlib.pyplot as plt`

`# Place your username and token here`
`username = 'Your_Username_here' `
`token = 'Your_Token_here' `

`# Download AERIoe data and ceilometer data`
`act.discovery.download_data(username, token, 'sgpaerioe1turnC1.c1', '2022-02-11', '2022-02-11') `
`act.discovery.download_data(username, token, 'sgpceilC1.b1', '2022-02-11', '2022-02-11') `

`# Read in AERIoe and ceilometer data`
`aerioe_ds = act.io.armfiles.read_netcdf('sgpaerioe1turnC1.c1/sgpaerioe1turnC1.c1.20220211.000354.nc')`
`ceil_ds = act.io.armfiles.read_netcdf('sgpceilC1.b1/sgpceilC1.b1.20220211.000005.nc')`

`# There isn't information content from the AERI above 3 km`
`# Remove data with a height above 3 km`
`aerioe_ds = aerioe_ds.sel(height=aerioe_ds.coords['height'] <= 3)`

`# Convert Ceilometer cloud base height to km`
`ceil_ds['first_cbh'] = ceil_ds['first_cbh']/1000`

`# Remove first_cbh if it is higher than 3 km`
`ceil_ds['first_cbh'] = ceil_ds['first_cbh'][~(ceil_ds['first_cbh'] > 3)]`

`# Create a TimeSeriesDisplay object`
`display = act.plotting.TimeSeriesDisplay(`
`   {'AERIoe': aerioe_ds, 'Ceilometer': ceil_ds},`
`   subplot_shape=(2,), figsize=(20,10)`
`)`

`# Plot data`
`display.plot('first_cbh', dsname='Ceilometer', marker='+', color='black', markeredgewidth=3,`
`   linewidth=0, subplot_index=(0,), label='cbh')`
`display.plot('temperature', dsname='AERIoe', cmap='viridis', set_shading='nearest',`
`   add_nan=True, subplot_index=(0,))`

`display.plot('first_cbh', dsname='Ceilometer', marker='+', color='black', markeredgewidth=3,`
`   linewidth=0, subplot_index=(1,), label='cbh')`
`display.plot('waterVapor', dsname='AERIoe', cmap='act_HomeyerRainbow', set_shading='nearest',`
`   add_nan=True, subplot_index=(1,))`

`plt.savefig('sgpaerioe1turnC1.c1.20220211.png')`

**Comment by zssherman** [comment]
@dennyh-ssec Awesome plot! Thanks for sharing! Would you consider submitting this as a blog post or an example?

**Reply by dennyh-ssec** [reply]
How do I go about submitting it as an example?

Found another change that should be made.
```
ceil_ds.utils.change_units(variables='first_cbh', desired_unit='km')
#ceil_ds['first_cbh'] = ceil_ds['first_cbh']/1000
```

**Reply by zssherman** [reply]
Just for clarification your showcasing a workflow? Or are you encountering a problem in ACT? Just wanted to make sure. For examples you would name the example plot_yourexamplename in the examples directory. And follow a template similar to here:
https://github.com/ARM-DOE/ACT/blob/main/examples/workflows/plot_multiple_dataset.py

**Comment by AdamTheisen** [comment]
@dennyh-ssec just as a note on the example @zssherman pointed to.  You could follow that template without the need to add the data files to the act.test area.  The example would then just run once merged so we would need to test it out ahead of time but that would somewhat streamline it.  If you need help, let us know!  Either way it would be great to get you as an author on a pull request!

**Comment by zssherman** [comment]
@dennyh-ssec Let me know if you need help with this.

**Reply by dennyh-ssec** [reply]
@zssherman Yeah, I could use some help as I have never submitted anything to github before. Usually just work on our SSEC gitlab instance. I created a 'dennyh-ssec' branch and put the file in the ACT/example/workflows directory and tried to push the branch but ran into an error.

**Reply by dennyh-ssec** [reply]
[dennyh@armada workflows]$ git push origin dennyh-ssec
Username for 'https://github.com': dennyh-ssec
Password for 'https://dennyh-ssec@github.com': 
remote: Permission to ARM-DOE/ACT.git denied to dennyh-ssec.
fatal: unable to access 'https://github.com/ARM-DOE/ACT/': The requested URL returned error: 403

**Reply by zssherman** [reply]
Hey @dennyh-ssec  ! Good to see this moving forward. Can you check your remotes with:
```git remote -v``` you might need to set your origin and upstream branches. When you do your git push, i believe its pushing to the main repo which could be the issue.

**Comment by AdamTheisen** [comment]
Great to have this example up in the gallery! 
https://arm-doe.github.io/ACT/source/auto_examples/workflows/plot_aerioe_with_cbh.html#sphx-glr-source-auto-examples-workflows-plot-aerioe-with-cbh-py
